fix: forceNewDeployment to be a boolean.

yehudacohen opened the pull request:
*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

paragbhingre commented:
@yehuda-elementryx  Thank you for fixing this issue, could you please try switching the conditions like 
`( forceNewDeployInput || forceNewDeployInput.toLowerCase() === 'true' )` and see if the build succeeds?

dili91 commented on index.js:
```diff
@@ -224,7 +224,7 @@ async function run() {
 
     const forceNewDeployInput = core.getInput('force-new-deployment', { required: false }) || false;
-    const forceNewDeployment = forceNewDeployInput != undefined && (forceNewDeployInput.toLowerCase === 'true' || forceNewDeployInput);
+    const forceNewDeployment = forceNewDeployInput != undefined && (forceNewDeployInput.toLowerCase() === 'true' || forceNewDeployInput);
```
To be honest I'd keep it simple and only accept boolean values here. This is what AWS SDK does and would be consistent with **wait-for-service-stability** param management

dili91 replied:
```suggestion
    const forceNewDeployment = core.getInput('force-new-deployment', { required: false }) || false;
```

dili91 commented on index.js:
```diff
@@ -224,7 +224,7 @@ async function run() {
     }
 
     const forceNewDeployInput = core.getInput('force-new-deployment', { required: false }) || false;
```
```suggestion
    
```

paragbhingre reviewed (commented):
@yehuda-elementryx Thank you for fixing this issue, could you please try switching the conditions like
( forceNewDeployInput || forceNewDeployInput.toLowerCase() === 'true' ) and see if the build succeeds?

dili91 commented:
> @yehuda-elementryx Thank you for fixing this issue, could you please try switching the conditions like
> ( forceNewDeployInput || forceNewDeployInput.toLowerCase() === 'true' ) and see if the build succeeds?

I'm neither a JS ninja nor a fan 😆 but this should work. We would accept every string values though
force-new-deployment: true => OK 
force-new-deployment: "true" => OK 
force-new-deployment: "whatever" => validation error from AWS SDK. 

Is it really wortwhile to add that `forceNewDeployInput.toLowerCase() === 'true'` just to cast `true` strings ? tbh I'd remove any validation and manipulation on this input, and let the SDK fail if it's the case. this is what I was trying to say with my previous comment :) 

That said, thanks again for looking into this @yehuda-elementryx

yehudacohen commented:
Hi all, apparently I hadn't packaged the code in the previous pull request. cleaned it up slightly, fixed failing unit tests and pushed. Should be good to go now. Tested from a separate repo to ensure everything was working as expected.

yehudacohen commented:
> > @yehuda-elementryx Thank you for fixing this issue, could you please try switching the conditions like
> > ( forceNewDeployInput || forceNewDeployInput.toLowerCase() === 'true' ) and see if the build succeeds?
> 
> I'm neither a JS ninja nor a fan 😆 but this should work. We would accept every string values though
> force-new-deployment: true => OK
> force-new-deployment: "true" => OK
> force-new-deployment: "whatever" => validation error from AWS SDK.
> 
> Is it really wortwhile to add that `forceNewDeployInput.toLowerCase() === 'true'` just to cast `true` strings ? tbh I'd remove any validation and manipulation on this input, and let the SDK fail if it's the case. this is what I was trying to say with my previous comment :)
> 
> That said, thanks again for looking into this @yehuda-elementryx

Github actions and boolean properties don't play nicely together. Take a look at this: https://github.com/actions/toolkit/issues/361

paragbhingre commented:
@yehudacohen I looked at your code and your input.test.js file also sends only string values. I'm still wondering how your code will take care of boolean values? When I tried running your code, my tests fail as I send actual boolean instead of string.

yehudacohen commented:
@paragbhingre there are no boolean values that can be return from the `core.getInput()` function. See here: https://github.com/actions/toolkit/issues/361

paragbhingre commented on dist/index.js:
```diff
@@ -388,8 +388,8 @@ async function run() {
 
-    const forceNewDeployInput = core.getInput('force-new-deployment', { required: false }) || false;
-    const forceNewDeployment = forceNewDeployInput != undefined && (forceNewDeployInput.toLowerCase === 'true' || forceNewDeployInput);
+    const forceNewDeployInput = core.getInput('force-new-deployment', { required: false }) || 'false';
```
Why do we need `|| 'false'` condition? Can't we just have the same logic as it is done for the **wait-for-service-stability** parameter?

```
const forceNewDeployInput = core.getInput('force-new-deployment', { required: false });
const forceNewDeployment = forceNewDeployInput.toLowerCase() === 'true';
```

yehudacohen replied:
Technically it's not necessary, but I find it easier to read. This is because `core.getInput('force-new-deployment', { required: false})` returns an empty string if it's not defined. To a reader who isn't familiar with the API, this makes sure they understand that if it isn't defined, the response should still be handled as a string rather than as an `undefined` or `null` value.

If you need it removed to merge the PR I'll remove it, but I find an explicit string value of `'false'` cleaner than an expectation that the `core.getInput()` function returns an empty string when the input is not provided.

paragbhingre replied:
@yehudacohen Great. I guess that extra condition works for me. Thank you for fixing the issue.

paragbhingre commented:
> @paragbhingre there are no boolean values that can be returned from the `core.getInput()` function. See here: [actions/toolkit#361](https://github.com/actions/toolkit/issues/361)
> 
> Per that issue, all boolean values will be returned as strings.

Thanks for the clarification @yehudacohen. I just have one last question that why do we have

> @paragbhingre there are no boolean values that can be returned from the `core.getInput()` function. See here: [actions/toolkit#361](https://github.com/actions/toolkit/issues/361)
> 
> Per that issue, all boolean values will be returned as strings.

Thank you for your clarification @yehudacohen, I just have one last question that I have asked in the following comment.
